hive/replays/on_policy_replay.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+
+from hive.replays.circular_replay import CircularReplayBuffer
+from hive.utils.advantage_computation_utils import AdvantageComputationFn
+
+
+class OnPolicyReplayBuffer(CircularReplayBuffer):
+    """An extension of the CircularReplayBuffer for on-policy learning algorithms"""
+
+    def __init__(
+        self,
+        capacity: int = 10000,
+        stack_size: int = 1,
+        n_step: int = 1,
+        gamma: float = 0.99,
+        compute_advantage_fn: AdvantageComputationFn = None,
+        observation_shape=(),
+        observation_dtype=np.uint8,
+        action_shape=(),
+        action_dtype=np.int8,
+        reward_shape=(),
+        reward_dtype=np.float32,
+        extra_storage_types=None,
+        num_players_sharing_buffer: int = None,
```

Comment:
Not sure this works with multiple agents. Maybe remove this from the parameter list.